tsconfig.json

```diff
@@ -0,0 +1,19 @@
+{
+    "compilerOptions": {
+      "baseUrl": ".",
+      "declarationDir": "dist/types",
+      "outDir": "dist/lib",
+      "preserveConstEnums": true,
+      "isolatedModules": true,
+      "sourceMap": true,
+      "declaration": true,
+      // Needed for refactor, change&remove like comments bellow suggest
+      "allowJs": true, // remove
+      "noImplicitAny": false, // true
+      "types": ["mocha", "node"], // remove
+      "esModuleInterop": true, // remove
+      "allowSyntheticDefaultImports": true, // check
+    },
+    "include": ["src/**/*", "test/**/*"],
+    "exclude": ["node_modules", "dist"]
+  }
```

Comment:
Please fix this unexpected tab before  this line ;D

Btw. I have TS warning:

![image](https://user-images.githubusercontent.com/15138956/150125725-cebc0125-52c4-46c9-847a-cfdda5555af0.png)